### PR TITLE
#606 boolean graph type not converting numerical string

### PR DESCRIPTION
**Pull request by KoditkarVedant**
- [x] Add ability to convert `"1"` to `"0"` to `true` and `false` respectively
- [x] Change `Serialize` for `BooleanGraphType` to call `ParseValue`
- [x] Added Test cases for changes

### CODE REVIEW

**Comment by KoditkarVedant** [comment]
Hello @joemcbride, Can you please look into this PR? Let me know if there are any changes I need to make.

File: src/GraphQL/Types/BooleanGraphType.cs [hunk 1]
```diff
@@ -11,10 +12,11 @@ public BooleanGraphType()
 
         public override object Serialize(object value)
         {
-            if (value is bool)
+            try
             {
-                return (bool) value;
+                return ParseValue(value);
             }
+            catch(Exception) { }
```
Comment (joemcbride) [inline src/GraphQL/Types/BooleanGraphType.cs]: Let’s not catch any exceptions here and let them bubble up.

File: src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs [hunk 2]
```diff
@@ -43,5 +44,46 @@ public void coerces_string_True()
...
+            type.ParseValue("0").ShouldBe(false);
+        }
+
+        [Theory]
+        [InlineData("abc")]
+        [InlineData("21")]
+        public void coerces_input_to_exception(string input)
+        {
+            FormatException formatException = Assert.Throws<FormatException>(() => type.ParseValue(input));
+            Assert.Equal("String was not recognized as a valid Boolean.", formatException.Message);
```
Comment (joemcbride) [inline src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs]: Can you switch this to formatException.Message.ShouldBe(...)

File: src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs [hunk 3]
```diff
@@ -43,5 +44,46 @@ public void coerces_string_True()
...
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData("0")]
+        [InlineData("False")]
+        [InlineData("false")]
```
Comment (joemcbride) [inline src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs]: For completeness can you add an actual Boolean value?

File: src/GraphQL/Types/BooleanGraphType.cs (hunk 1 above)
Comment (KoditkarVedant) [reply src/GraphQL/Types/BooleanGraphType.cs]: I had caught the exception because the old Serialize method was always returning the boolean value.

Comment (KoditkarVedant) [reply src/GraphQL/Types/BooleanGraphType.cs]: Done.

File: src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs (hunk 2 above)
Comment (KoditkarVedant) [reply src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs]: Done.

File: src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs (hunk 3 above)
Comment (KoditkarVedant) [reply src/GraphQL.Tests/Types/BooleanGraphTypeTests.cs]: Done.

**Comment by KoditkarVedant** [comment]
Hello @joemcbride, completed the review changes.

**Comment by joemcbride** [comment]
@KoditkarVedant Thanks for this!

**Pull request merged**